pkg/driver/node.go

```diff
@@ -848,6 +849,31 @@ func startNotReadyTaintWatcher(clientset kubernetes.Interface, maxWatchDuration
 		return
 	}
 
+	var mutex sync.Mutex
+	// Add additional 90 seconds to the context for the last-try taint removal
+	ctx, cancel := context.WithTimeout(context.Background(), maxWatchDuration+90*time.Second)
+	defer cancel()
+
+	canPatchNode, err := clientset.AuthorizationV1().
+		SelfSubjectAccessReviews().
+		Create(ctx, &authv1.SelfSubjectAccessReview{
+			Spec: authv1.SelfSubjectAccessReviewSpec{
+				ResourceAttributes: &authv1.ResourceAttributes{
+					Verb:     "patch",
+					Resource: "nodes",
+					Name:     nodeName,
+				},
+			},
+		}, metav1.CreateOptions{})
+	if err != nil {
+		klog.V(6).InfoS("Failed to check node update permissions", "err", err)
+		// Silently proceed here, in case we have permission, worst case scenario is log messages
```

Comment:
What is the value in adding this extra code path to check for auth permission up front? afaik, the API returns a clear error message when unable to patch. In other words, is there a strong reason not to always make the request and let the server handle auth?